Lib/cu2qu/rf.py

```diff
@@ -46,6 +44,103 @@ def zip(*args):
     return _zip(*args)
 
 
+class GetSegmentsPen(AbstractPen):
+    """Pen to collect segments into lists of points for conversion."""
+
+    def __init__(self):
+        self._last_pt = None
+        self.segments = []
+
+    def _add_segment(self, tag, *args):
+        self.segments.append((tag, args))
+
+    def moveTo(self, pt):
+        self._add_segment('move', pt)
+        self._last_pt = pt
+
+    def lineTo(self, pt):
+        self._add_segment('line', pt)
+        self._last_pt = pt
+
+    def curveTo(self, *points):
+        self._add_segment('curve', self._last_pt, *points)
```

Comment:
Well, half the calls to _add_segment don't include points (close, end, component), but if it'd be more sound we could indeed check that the tag is in ['move', 'line', 'curve'] and if so add the last point in _add_segment.